### PR TITLE
Integration with k3s + argo workflow

### PR DESCRIPTION
**Pull request by ktock**
This attempts to integrate eStargz + k3s + Argo Workflow with a simple example Golang CI.

### CODE REVIEW

**Comment by ktock** [comment]
Maybe we can add this as a demo doc rather than tests.

**Comment by ktock** [comment]
Elapsed time of the workflow shortened 92s -> 50s on average.

https://github.com/containerd/stargz-snapshotter/pull/449/checks?check_run_id=3516746589

instance info:

```json
{
  "location": "eastus",
  "vmSize": "Standard_DS2_v2"
}
```

workflow elapsed time:

```json
{"type" : "org", "snapshotter" : "overlayfs", "elapsed" : "93"}
{"type" : "esgz", "snapshotter" : "stargz", "elapsed" : "51"}
{"type" : "org", "snapshotter" : "overlayfs", "elapsed" : "92"}
{"type" : "esgz", "snapshotter" : "stargz", "elapsed" : "51"}
{"type" : "org", "snapshotter" : "overlayfs", "elapsed" : "93"}
{"type" : "esgz", "snapshotter" : "stargz", "elapsed" : "50"}
```

**Comment by ktock** [comment]
cc @AkihiroSuda

File: script/k3s-argo-workflow/run.sh [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+      args: ["
+        uname -a ;
+        cat /etc/os-release ;
+        /go/src/github.com/golang/example/hello/hello
+      "]
+      volumeMounts:
+      - name: workdir
+        mountPath: /go
+EOF
+}
```
Comment (AkihiroSuda) [inline script/k3s-argo-workflow/run.sh:200]: The yaml looks too big to embed in shell.

Just split this yaml as a yaml file, and use envsubst or erb for templating.

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**